kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java

```diff
@@ -98,9 +99,18 @@ public Optional<CheckpointMetaData> readLastCheckpointFile(TableClient tableClie
      */
     private Optional<CheckpointMetaData> loadMetadataFromFile(TableClient tableClient) {
         try {
-            // For now we use file size = 0 and modification time = 0, in the future we should use
-            // listFrom to retrieve the real values see delta-io/delta#2140
-            FileStatus lastCheckpointFile = FileStatus.of(lastCheckpointFilePath.toString(), 0, 0);
+            FileSystemClient fileSystemClient = tableClient.getFileSystemClient();
+            FileStatus lastCheckpointFile = null;
+            try (CloseableIterator<FileStatus> lastCheckpointFileStatusIterator =
+                fileSystemClient.listFrom(
+                    this.lastCheckpointFilePath.toString()
+                )
+            ) {
+                while (lastCheckpointFileStatusIterator.hasNext()) {
+                    lastCheckpointFile = lastCheckpointFileStatusIterator.next();
+                }
+            }
+
```

Comment:
Not directly related: can you please add ` TODO: log the exception, once https://github.com/delta-io/delta/issues/2230 is resolved` in the `catch(Exception ex)` block?

Reply:
Added